File: algorithms/2_medium/M/03001_minimum_moves_to_capture_the_queen/o1.py
class Solution:
    def minMovesToCaptureTheQueen(self, a: int, b: int, c: int, d: int, e: int, f: int) -> int:
        if a == e:
            if c == a and d > min(b, f) and d < max(b, f):
                return 2
            return 1

        if b == f:
            if b == d and c > min(a, e) and c < max(a, e):
                return 2
            return 1

        if abs(e - c) == abs(f - d):
            if abs(a - c) == abs(b - d):
                if a > min(c, e) and a < max(c, e) and b > min(d, f) and b < max(d, f):
                    return 2
            return 1

        return 2

File: algorithms/2_medium/M/03001_minimum_moves_to_capture_the_queen/o1.rs
impl Solution {
    pub fn min_moves_to_capture_the_queen(a: i32, b: i32, c: i32, d: i32, e: i32, f: i32) -> i32 {
        if a == e {
            if c == a && d > b.min(f) && d < b.max(f) {
                return 2;
            }

            return 1;
        }

        if b == f {
            if b == d && c > a.min(e) && c < a.max(e) {
                return 2;
            }

            return 1;
        }

        if (e - c).abs() == (f - d).abs() {
            if (a - c).abs() == (b - d).abs() {
                if a > c.min(e) && a < c.max(e) && b > d.min(f) && b < d.max(f) {
                    return 2;
                }
            }

            return 1;
        }

        2
    }
}
